[BUG] Tests failing due to retiring of torch.testing.assert_allclose

## Describe the bug

Due to [this change](https://github.com/pytorch/pytorch/pull/87974), `torch.testing.assert_allclose` has been removed which is causing unit tests to fail.

[Output from a recent CI run](https://app.circleci.com/pipelines/github/pytorch/rl/4214/workflows/9f006a20-055f-4a7a-83ea-d9d8b3fbc65c/jobs/98683)
```bash
FAILED test/test_rb.py::test_prototype_prb[device0-True-pk] - AttributeError: module 'torch.testing' has no attribute 'assert_allclose'
FAILED test/test_rb.py::test_prototype_prb[device0-True-td_error] - AttributeError: module 'torch.testing' has no attribute 'assert_allclose'
FAILED test/test_rb.py::test_prototype_prb[device0-False-pk] - AttributeError: module 'torch.testing' has no attribute 'assert_allclose'
FAILED test/test_rb.py::test_prototype_prb[device0-False-td_error] - AttributeError: module 'torch.testing' has no attribute 'assert_allclose'
```

## To Reproduce

Run `pytest` with pytorch nightly build installed. 

## Reason and Possible fixes

`torch.testing.assert_close` should be a drop-in replacement; [reference](https://github.com/pytorch/pytorch/issues/61844)

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

--- a/test/test_rb.py
+++ b/test/test_rb.py
@@ -236,16 +236,12 @@ def test_prototype_prb(priority_key, contiguous, device):
     rb.update_tensordict_priority(s)
     s, _ = rb.sample(5)
     assert (val == s.get("a")).sum() >= 1
-    torch.testing.assert_allclose(
-        td2[idx0].get("a").view(1), s.get("a").unique().view(1)
-    )
+    torch.testing.assert_close(td2[idx0].get("a").view(1), s.get("a").unique().view(1))
 
     # test updating values of original td
     td2.set_("a", torch.ones_like(td2.get("a")))
     s, _ = rb.sample(5)
-    torch.testing.assert_allclose(
-        td2[idx0].get("a").view(1), s.get("a").unique().view(1)
-    )
+    torch.testing.assert_close(td2[idx0].get("a").view(1), s.get("a").unique().view(1))
 
 
 @pytest.mark.parametrize("stack", [False, True])

